### PR TITLE
feat(ilp): disconnect on error flag

### LINKED ISSUE #1823: Add config option to close TCP connection on an invalid ILP row
### Is your feature request related to a problem?

Current ILP implementation lacks acknowledgments. So, if an application sends invalid data, the problematic rows get silently ignored and the only way to learn this is to check server logs.

To simplify UX in development environments, we could add a new config option that would tell QuestDB to close the TCP connection once an invalid row is received or an error occurred when inserting a row. 

### Describe the solution you'd like.

_No response_

### Describe alternatives you've considered.

_No response_

### Additional context.

_No response_

### PR DESCRIPTION
**Pull request by glasstiger**
Introducing a new boolean config option for ILP: `line.tcp.disconnect.on.error`
If for any reason ILP cannot process a message and this flag is true the connection will be closed and ingestion stops.
If the flag is false the problematic message is ignored and ingestion continues.
More details in the linked issue.
Closes #1823

This PR also includes tests for ILP type conversions.
It demonstrates how different ILP types can fit into QuestDB types, what is a valid conversion and what is not.
This will be also documented in the official QuestDB docs.
[Docs PR](https://github.com/questdb/questdb.io/pull/710)

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 14 / 15 (93.33%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpConnectionContext.java|9|10|90.00%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/TableUpdateDetails.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/DefaultLineTcpReceiverConfiguration.java|1|1|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 18 / 19 (94.74%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpConnectionContext.java|9|10|90.00%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/TableUpdateDetails.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/DefaultLineTcpReceiverConfiguration.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementEvent.java|4|4|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 18 / 19 (94.74%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpConnectionContext.java|9|10|90.00%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/TableUpdateDetails.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/DefaultLineTcpReceiverConfiguration.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementEvent.java|4|4|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 18 / 19 (94.74%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpConnectionContext.java|9|10|90.00%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/TableUpdateDetails.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/DefaultLineTcpReceiverConfiguration.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementEvent.java|4|4|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 18 / 19 (94.74%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpConnectionContext.java|9|10|90.00%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/TableUpdateDetails.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/DefaultLineTcpReceiverConfiguration.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementEvent.java|4|4|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 18 / 19 (94.74%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpConnectionContext.java|9|10|90.00%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/TableUpdateDetails.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/DefaultLineTcpReceiverConfiguration.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementEvent.java|4|4|100.00%|

File: core/src/main/java/io/questdb/PropServerConfiguration.java [hunk 1]
```diff
@@ -824,6 +825,7 @@ public PropServerConfiguration(
                     this.lineTcpAuthDbPath = new File(root, this.lineTcpAuthDbPath).getAbsolutePath();
                 }
                 this.minIdleMsBeforeWriterRelease = getLong(properties, env, "line.tcp.min.idle.ms.before.writer.release", 10_000);
+                this.lineTcpDisconnectOnError = getBoolean(properties, env, "line.tcp.disconnect.on.error", false);
```
[the remark this answers is not in the document]
Comment (glasstiger) [reply core/src/main/java/io/questdb/PropServerConfiguration.java]: changed it

File: core/src/main/java/io/questdb/cutlass/line/tcp/LineTcpConnectionContext.java [hunk 2]
```diff
@@ -48,6 +48,7 @@
     private final DirectByteCharSequence byteCharSequence = new DirectByteCharSequence();
     private final LineTcpParser parser = new LineTcpParser();
     private final FloatingDirectCharSink floatingDirectCharSink = new FloatingDirectCharSink();
+    private final boolean shouldDisconnectOnError;
```
[the remark this answers is not in the document]
Comment (glasstiger) [reply core/src/main/java/io/questdb/cutlass/line/tcp/LineTcpConnectionContext.java]: removed "should"

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 17 / 19 (89.47%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/DefaultLineTcpReceiverConfiguration.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpConnectionContext.java|9|10|90.00%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/TableUpdateDetails.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementEvent.java|4|4|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 17 / 19 (89.47%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/DefaultLineTcpReceiverConfiguration.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpConnectionContext.java|9|10|90.00%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/TableUpdateDetails.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementEvent.java|4|4|100.00%|

**Review by bluestreak01 — APPROVED** [review]

**Pull request merged**